streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -683,20 +683,16 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
             _ <- self.foreachManaged { a =>
                   for {
                     latch <- Promise.make[Nothing, Unit]
+                    _     <- permits.acquire
```

Comment:
Let's add a comment that explains why an unbracketed `permits.acquire` here is safe - from my reasoning this is ok because the fiber executing this could only be interrupted if the entire stream is interrupted, and at that point it doesn't matter if we leak an acquisition.

Reply:
You're right, it's safer and easier to reason about to put it back in the bracket, since now the latch protects the continuation of the foreachManaged. Making the change.

Reply:
But nothing is changed if you put it back, no? Did the latch not protect the continuation before?

Reply:
Let me sync my local repo first so I can observe the changes. I'll comment in a few minutes.

Reply:
Alright, I've no idea what's going on now...

Reply:
So `flatMapPar` must have been good, we just missed the test that proves it, so I guess that's solved now.

Reply:
`flatMapParSwitch` and the `cancelersGuard` trick must have added non-determinism by waking up fibers out of order...